libnavigator/src/main/java/com/mapbox/navigation/navigator/NavigationOkHttpService.kt

```diff
@@ -0,0 +1,169 @@
+package com.mapbox.navigation.navigator
+
+import com.mapbox.base.common.logger.Logger
+import com.mapbox.base.common.logger.model.Message
+import com.mapbox.base.common.logger.model.Tag
+import com.mapbox.bindgen.Expected
+import com.mapbox.bindgen.ExpectedFactory
+import com.mapbox.common.HttpRequest
+import com.mapbox.common.HttpRequestError
+import com.mapbox.common.HttpRequestErrorType
+import com.mapbox.common.HttpResponse
+import com.mapbox.common.HttpResponseCallback
+import com.mapbox.common.HttpResponseData
+import com.mapbox.common.HttpServiceInterface
+import java.io.IOException
+import java.util.HashMap
+import java.util.Locale
+import java.util.UUID
+import java.util.concurrent.locks.ReentrantLock
+import okhttp3.Call
+import okhttp3.Callback
+import okhttp3.Headers
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+import okhttp3.logging.HttpLoggingInterceptor
+
+internal class NavigationOkHttpService(
+    private val httpClientBuilder: OkHttpClient.Builder,
+    private val logger: Logger? = null
+) : HttpServiceInterface() {
+
+    companion object {
+        internal const val HEADER_ENCODING = "Accept-Encoding"
+        internal const val GZIP = "gzip, deflate"
+        private const val LOG_TAG = "NavigationOkHttpService"
+    }
+
+    private val lock = ReentrantLock()
+
+    private val callMap = hashMapOf<Long, Call>()
+
+    private val httpClient: OkHttpClient by lazy {
+        if (BuildConfig.DEBUG) {
+            val interceptor = HttpLoggingInterceptor(
+                HttpLoggingInterceptor.Logger { message ->
+                    logger?.d(
+                        tag = Tag(LOG_TAG),
+                        msg = Message(message)
+                    )
+                }
+            ).setLevel(HttpLoggingInterceptor.Level.BASIC)
+
+            httpClientBuilder.addInterceptor(interceptor)
+        }
+        httpClientBuilder.build()
```

Comment:
If we would OkHttpService to `base` and we are relying on a binary, the buildConfig would always be release and logging would only be available when running inside base project. Not a blocker but just fyi.

Reply:
Yea, that's usable for the nav project by we can remove it when moving upstream.